components/compass-runtime-agent/cmd/main.go

```diff
@@ -114,10 +113,40 @@ func main() {
 		log.Error("Unable to initialize Compass Connection CR")
 	}
 
-	// Start the Cmd
-	log.Info("Starting the Cmd.")
-	if err := mgr.Start(signals.SetupSignalHandler()); err != nil {
-		log.Error(err, "Unable to run the manager")
-		os.Exit(1)
+	// Initialize metrics logger
+	log.Infoln("Initializing metrics logger")
+	metricsLogger, err := newMetricsLogger(options.MetricsLoggingTimeInterval)
+	if err != nil {
+		log.Error(errors.Wrap(err, "Unable to create metrics logger"))
 	}
+
+	runManagerAndLoggerConcurrently(mgr, metricsLogger)
+	os.Exit(1)
+}
+
+func runManagerAndLoggerConcurrently(manager manager.Manager, logger metrics.Logger) {
```

Comment:
I suppose it is possible that logger maybe passed as nil here - after newMetricsLogger function failed (for example when creation of Resource fetcher has failed).  In such a case logger.Log(quitChannel) (line 146) will crash

Reply:
Good catch! Thanks, I'll fix it ;>